modules/search/class.jetpack-search-helpers.php

```diff
@@ -738,4 +738,28 @@ public static function generate_post_type_customizer_ids() {
 			get_post_types( array( 'exclude_from_search' => false ), 'objects' )
 		);
 	}
+
+	/**
+	 * Sanitizes a checkbox value for writing to the database.
+	 *
+	 * @since 8.9.0
+	 *
+	 * @param any $value from the customizer form.
+	 * @return string either '0' or '1'.
+	 */
+	public static function sanitize_checkbox_value( $value ) {
+		return true === $value ? '1' : '0';
```

Comment:
Do we need this to be `true == $value` for backwards compatibility or some other cases?

It feels like we should make sure that '1' will match against true here even though I am not sure how that case will happen.

Reply:
This shouldn't be necessary; this function sanitizes Customizer form values -- not the saved configuration values from the database. Checkbox inputs from the Customizer form seem to always return a boolean.

The use-case you've described is actually handled by `sanitize_checkbox_value_for_js`, which sanitizes the saved value from the database before being used to initialize the Customizer form. This function has explicit handling for the `'1'` value scenario you've mentioned.

Reply:
OK, got it. thanks for the explanation.